File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    let ret = [];
    let mode = 0
    
    for(let i=0; i<code.length; i++) {
        if(code[i] == 1) {
            mode = 1 - mode;
        } else {
            if((mode === 0 && i%2 === 0) || (mode === 1 && i%2 === 1)) {
                ret.push(code[i])
            }
        }
    }
    
    return ret.length ? ret.join('') : 'EMPTY';
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        int mode = 0;
        StringBuilder ret = new StringBuilder();
        
        for (int i = 0; i < code.length(); i++) {
            char c = code.charAt(i);
            
            if (c == '1') {
                mode = 1 - mode;
                continue;
            }
            
            if ((mode == 0 && i % 2 == 0) || (mode == 1 && i % 2 == 1)) {
                ret.append(c);
            }
        }
        
        return ret.length() > 0 ? ret.toString() : "EMPTY";
    }
}